backend/graphql/types/residentType.ts

```diff
@@ -0,0 +1,78 @@
+import { gql } from "apollo-server-express";
+
+// TODO: Look into custom types for dates and date time types
+
+const residentType = gql`
+  type ResidentDTO {
+    userId: Int!
+    residentId: Int!
+    email: String!
+    phoneNumber: String
+    firstName: String!
+    lastName: String!
+    displayName: String
+    profilePictureURL: String
+    isActive: Boolean!
+    birthDate: Date!
+    roomNumber: Int!
+    credits: Float!
+    dateJoined: Date!
+    dateLeft: Date
+    notes: String
+  }
+
+  input CreateResidentDTO {
+    email: String!
+    password: String!
+    phoneNumber: String
+    firstName: String!
+    lastName: String!
+    displayName: String
+    profilePictureURL: String
+    residentId: Int!
+    birthDate: Date!
+    roomNumber: Int!
+    credits: Float!
+    dateJoined: Date!
+    dateLeft: Date
+    notes: String
+  }
+
+  input UpdateResidentDTO {
```

Comment:
Should there be a way to update a resident to be inActive? There isn't a field for the isActive column in this dto and I was wondering if that was intended

Reply:
Good point. We can address this in a separate ticket, filed one here: https://www.notion.so/uwblueprintexecs/Create-mutations-to-set-resident-or-staff-to-inactive-bcc84f2c1e75480c96002e5ad7ccaef5?pvs=4